Hippo.Web/Controllers/OrderController.cs

```diff
@@ -0,0 +1,866 @@
+            _historyService = historyService;
+        }
+
+
+
+        [HttpGet]
+        [Route("api/order/validateChartString/{chartString}/{direction}")]
+        public async Task<ChartStringValidationModel> ValidateChartString(string chartString, string direction)
+        {
+            return await _aggieEnterpriseService.IsChartStringValid(chartString, direction);
+
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> MyOrders()
+        {
+            var currentUser = await _userService.GetCurrentUser();
+
+            var currentUserAccount = await _dbContext.Accounts.SingleOrDefaultAsync(a => a.Cluster.Name == Cluster && a.OwnerId == currentUser.Id);
+            if (currentUserAccount == null)
+            {
+                return Ok(new OrderListModel[0]);
+            }
+
+            var model = await _dbContext.Orders.Where(a => a.Cluster.Name == Cluster && a.PrincipalInvestigatorId == currentUserAccount.Id).Select(OrderListModel.Projection()).ToListAsync(); //Filter out inactive orders?
+
+            return Ok(model);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> AdminOrders()
+        {
+            var currentUser = await _userService.GetCurrentUser();
+            var permissions = await _userService.GetCurrentPermissionsAsync();
+            var isClusterOrSystemAdmin = permissions.IsClusterOrSystemAdmin(Cluster);
+
+            if (!isClusterOrSystemAdmin)
+            {
+                return BadRequest("You do not have permission to view this page.");
+            }
+            var currentUserAccount = await _dbContext.Accounts.SingleOrDefaultAsync(a => a.Cluster.Name == Cluster && a.OwnerId == currentUser.Id);
+            if (currentUserAccount == null)
+            {
+                return Ok(new OrderListModel[0]);
+            }
+
+            //Probably will want to filter out old ones that are completed and the expiration date has passed.
+            var adminStatuses = new List<string> { Order.Statuses.Submitted, Order.Statuses.Processing, Order.Statuses.Active, Order.Statuses.Completed };
+
+            var model = await _dbContext.Orders.Where(a => a.Cluster.Name == Cluster && adminStatuses.Contains(a.Status)).Select(OrderListModel.Projection()).ToListAsync(); //Filter out inactive orders?
+
+            return Ok(model);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> Get(int id)
+        {
+            //TODO: When accounts has an IsActive flag, we will need to ignore the query filters.
+
+            var model = await _dbContext.Orders.Where(a => a.Cluster.Name == Cluster && a.Id == id)
+                .Include(a => a.MetaData).Include(a => a.Payments).Include(a => a.PrincipalInvestigator).ThenInclude(a => a.Owner)
+                .Include(a => a.History.Where(w => w.Type == History.HistoryTypes.Primary)).ThenInclude(a => a.ActedBy)
+                .Select(OrderDetailModel.Projection())
```

Comment:
Include does nothing and is not necessary if you are doing a projection.  remove all includes.  Also, it's not a big deal but I don't think getting an order should by default include all history.  Seems like that should be a separate call, no?

Reply:
I think it is only getting the primary history when I do it like this and not all of it?
Anyway, will take a look. I see to remember having a problem...

Reply:
Maybe I did it without projection first... Will fix other places too